src/sql/workbench/services/connection/browser/connectionDialogService.ts

```diff
@@ -535,12 +539,21 @@ export class ConnectionDialogService implements IConnectionDialogService {
 		actions.push(new Action('Diagnose', localize('diagnoseError', "Diagnose Error"), undefined, true, async () => {
 			this._connectionDialog.close();
 			this.telemetryService.createActionEvent(TelemetryKeys.TelemetryView.ConnectionErrorDialog, TelemetryKeys.TelemetryAction.Click, 'DiagnoseError');
+			this.showTroubleshooterDialog(headerTitle, message, messageDetails, []);
```

Comment:
Can you move the below to MSSQL extension? Since the diagnosis error button will only be shown for MSSQL provider. Which means, this piece of code, and call to show troubleshooting dialog should be triggered from MSSQL when it handles errors.

You can design the dialog in SQL Workbench but it will be triggered from MSSQL extension. Look at "ErrorDiagnosticsProvider.handleConnectionError" and how it's called from SQL Workbench.